core/trino-main/src/test/java/io/trino/operator/GroupByHashYieldAssertion.java

```diff
@@ -187,10 +187,16 @@ public static GroupByHashYieldResult finishOperatorWithYieldingGroupByHash(List<
 
                 // Assert the estimated reserved memory before rehash is very close to the one after rehash
                 long rehashedMemoryUsage = operator.getOperatorContext().getDriverContext().getMemoryUsage();
-                assertBetweenInclusive(rehashedMemoryUsage * 1.0 / newMemoryUsage, 0.99, 1.01);
+                // Usually the error is < 1%, but it can be up to 5% because MultiChannelGroupByHash.getEstimatedSize
+                // accounts for changes in completedPagesMemorySize, which is increased if new page is
+                // added by addNewGroup (an even that cannot be predicted as it depends on the number of unique groups
```

Comment:
Why this test is non-deterministic?

Reply:
This check is inside a loop over input pages. Some input that triggers rehash will start a new page and some will not. This could be avoided by using input pages with a large enough count of unique values but that would always make the error ~5%.
On top of that, this is actually a utility test/assertion used by multiple tests.

Reply:
> This check is inside a loop over input pages. Some input that triggers rehash will start a new page and some will not.

Previously a new page could have also been started when page is full.

Reply:
> Previously a new page could have also been started when page is full.

Yes but not necessarily during rehash. I'm not 100 % sure this was the case. Will double chek

Reply:
It shouldn't happen during rehash in new code too

Reply:
I checked and as I expected, in the previous version, the `startNewPage` happens only during processing a page without rehash.
This seems to be a coincidence based on the size of the full `currentPageBuilder` and the memory usage of `MultuChannelGroupByHash` (both change with this PR).
Why do you think it shouldn't happen in the new code?